### PR TITLE
#169817814 User should login with email and Password

### PR DESCRIPTION
**Pull request by Jajabenit250**
#### What does this PR do?
- User Should SignIn with Email and Password
#### Description of Task to be completed?
- User Input his/ her Email and Password 
- If email and Password match to what is registered to our database user will get success message
- If the Input doesn't match to registered one user will get error message ("Email and password not match")
- User has to input valid Email and Password. e.g: (example@jaja.com, 123456790889).
#### How should this be manually tested?
-  Clone this repository
-  git checkout feature/169817814/user-login-email-password
-  open console and type and run(- npm install) to install all dependencies
-  and then run (- npm start) to start the server
#### Any background context you want to provide?
- User need to be verify his email before sign In 
#### What are the relevant pivotal tracker stories?
https://www.pivotaltracker.com/story/show/169817814
#### Screenshots (if appropriate) 
updated screenshot

### CODE REVIEW

File: __tests__/views/signin.view.test.js [hunk 1]
```diff
@@ -1,10 +1,23 @@
+// import React from 'react';
```
Comment (mujohn26) [inline __tests__/views/signin.view.test.js]: Good Job @Jajabenit250  but can you delete all these commented files if you are not using them.

File: __tests__/views/signin.view.test.js [hunk 2]
```diff
@@ -1,10 +1,23 @@
...
-import { mount } from 'enzyme';
-import Signin from '../../src/views/signin.view.jsx';
+import { shallow } from 'enzyme';
+import LoginPage from '../../src/views/signin.view'
+
+describe('Social Login component tests', () => {
+    it('should render SocialLogin component successfully', () => {
+        const wrapper = shallow(<LoginPage />);
+        expect(wrapper.find('div').length).toBe(1);
+        // expect(wrapper.find('a').length).toBe(2);
```
Comment (mujohn26) [inline __tests__/views/signin.view.test.js]: @Jajabenit250 delete these commented files.

File: src/components/App.jsx [hunk 3]
```diff
@@ -2,20 +2,19 @@ import React, { Component } from 'react';
...
-
+import { PrivateRoute } from '../components/auth/ProtectedRoute.jsx';
 class App extends Component {
 	render() {
 		return (
 			<div className='app'>
 				<Router>
 					<Switch>
-						<Route path='/' exact component={MainLayout} />
+						<PrivateRoute path='/' exact component={MainLayout} />
```
Comment (mujohn26) [inline src/components/App.jsx:12]: @Jajabenit250 Why did you changed this to private route.

**Review by mujohn26 — CHANGES_REQUESTED** [review]

File: src/components/App.jsx (hunk 3 above)
Comment (Jajabenit250) [reply src/components/App.jsx:12]: @mujohn26 To check if the user is already signed in and got to dashboard if not he will go to login instead.

**Review by mujohn26 — APPROVED** [review]

**Review by Ruqoyah — COMMENTED** [review]
Good Job! However, below are the feedback you need to work on

- When I visit localhost:8080, it redirects me to localhost:8080/auth/signin which is fine but the page doesn't show until I refresh the page again. Kindly fix that.

- I tried logging in using credentials that doesn't exist and it returned `user not found`, let's make the error message better by saying `invalid credentials` instead.

- Disable and add a loader to button (just like what Dominique did, you can reach out to him or test his branch to see what I'mm talking about) when a request is processing for good user experience.

- You save all the response from login to the localstorage as token instead of just saving the token alone.

Screenshot 2020-02-14 at 5 14 59 AM

- I got the above error from console after a successful login and it doesn't show the mainlayout page.


Note: Jordan is also working on the signin page, and I have given him feedback on what I noticed on his as well, I believe you guys should work together to get this fix so you can do the same thing. Whoever implement all the feedback first, I will merge his PR then the other person can get the update from develop.

**Comment by Ruqoyah** [comment]
Great Job

File: src/views/signin.view.jsx [hunk 4]
```diff
@@ -1,11 +1,64 @@
...
-        )
-    }
+export class SignIn extends PureComponent {
+	constructor(props) {
+		super(props);
+		this.state = {
+			redirect: false
+		};
+	}
+	  // componentDidUpdate(prevProps) {
```
Comment (Ruqoyah) [inline src/views/signin.view.jsx]: Please remove this commented code

File: src/views/signin.view.jsx [hunk 5]
```diff
@@ -1,11 +1,64 @@
 import React, { PureComponent } from 'react';
+import LoginForm from '../components/auth/Login.jsx';
+import { withStyles } from '@material-ui/core/styles';
+import Paper from '@material-ui/core/Paper';
+import Grid from '@material-ui/core/Grid';
+import Container from '@material-ui/core/Container';
+// import { useHistory } from 'react-router-dom';
+// import Footer from '../components/general'
```
Comment (Ruqoyah) [inline src/views/signin.view.jsx]: Remove it here as well

**Pull request merged**